test/util/framework/hcp_helper.go

```diff
@@ -245,3 +248,50 @@ func DeleteAllHCPClusters(
 
 	return nil
 }
+
+// VerifyNodePoolConfiguration verifies that a NodePool has the expected number of replicas and osDisk size.
+// This function uses the Kubernetes client to check the HyperShift NodePool CRD in the cluster namespace.
+func VerifyNodePoolConfiguration(ctx context.Context, adminRESTConfig *rest.Config, clusterName, nodePoolName string, expectedReplicas int32, expectedOsDiskSizeGiB int32) error {
+	dynamicClient, err := dynamic.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create dynamic client: %w", err)
+	}
+
+	nodePoolGVR := schema.GroupVersionResource{
+		Group:    "hypershift.openshift.io",
+		Version:  "v1beta1",
+		Resource: "nodepools",
+	}
+
+	namespace := clusterName
+	nodePool, err := dynamicClient.Resource(nodePoolGVR).Namespace(namespace).Get(ctx, nodePoolName, metav1.GetOptions{})
```

Comment:
I love seeing this verification using the kube API.  I'd love to see a general verification of
1. this many nodes exist
2. the nodes in the kube cluster are the ones in the azure API.

the specific details of for each arg could be verification funcs versus specific args.  See the verify cluster for inspiratino.